src/software/thunderscope/constants.py

```diff
@@ -335,6 +335,25 @@ class TrailValues:
     DEFAULT_TRAIL_LENGTH = 20
     DEFAULT_TRAIL_SAMPLING_RATE = 0
 
+
 class ProtoConfigurationConstant:
     DEFAULT_SAVE_DIRECTORY = "/tmp/tbotspython/thunderbots_configurations_proto"
```

Comment:
Actually I just realized that you're saving to `/tmp` which is not persisted between computer restart. You should be saving in `/opt/tbotspython/thunderbots_configurations_proto` instead

Reply:
that was intentional. since there is permission error if I were to put this in `/opt`. 

I think you would have to `chown -R` the directory in `setup.sh` if you were going to do that.

Reply:
How come we can save the layouts to /opt then? https://github.com/UBC-Thunderbots/Software/blob/207a68885932d3d2943cbbf33590ddfb0149c142/src/software/thunderscope/thunderscope.py#L157-L169

Reply:
If it works on your end, then I've no idea. It doesn't seem to work on my end. Here is the traceback. 

```
===== 2024.07.10 10:07:27 =====
Traceback (most recent call last):
  File "/home/vhe/.cache/bazel/_bazel_vhe/917dd48f905a75cc113dac63db2f3aa5/execroot/__main__/bazel-out
/k8-fastbuild/bin/software/thunderscope/thunderscope_main.runfiles/__main__/software/thunderscope/thun
derscope.py", line 140, in save_layout
    pathlib.Path(SAVED_LAYOUT_PATH).mkdir(exist_ok=True)
  File "/usr/lib/python3.8/pathlib.py", line 1288, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/opt/tbotspython/saved_tscope_layout'
===== 2024.07.10 10:07:27 =====
Traceback (most recent call last):
  File "/home/vhe/.cache/bazel/_bazel_vhe/917dd48f905a75cc113dac63db2f3aa5/execroot/__main__/bazel-out
/k8-fastbuild/bin/software/thunderscope/thunderscope_main.runfiles/__main__/software/thunderscope/thun
derscope.py", line 140, in save_layout
    pathlib.Path(SAVED_LAYOUT_PATH).mkdir(exist_ok=True)
  File "/usr/lib/python3.8/pathlib.py", line 1288, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/opt/tbotspython/saved_tscope_layout'

```

Reply:
Could you show me your `ls -l` in the `/opt` directory. 

For me, it shows the following: 

```
drwxr-xr-x  7 vhe  vhe  4096 Feb  1  1980 gradle-8.8
drwxr-xr-x 10 vhe  vhe  4096 Jun 18 18:38 idea
drwxr-xr-x 10 vhe  vhe  4096 Jun 26 22:23 meld-3.22.2
drwxr-xr-x  3 root root 4096 Oct  7  2023 microsoft
drwxr-xr-x  2 vhe  vhe  4096 Jul  2 18:04 shadowsocks
drwxr-xr-x  5 root root 4096 Jan 28 05:26 tbotspython
drwxrwxr-x  9 vhe  vhe  4096 Jun 19 21:32 tomcat
drwxr-xr-x  6 root root 4096 Jun 15 17:11 wine-stable
```

the folder need root permission I think as a result.

Reply:
Mine seems to be owned by my user and not root.
```
~/thunderbots/Software/src (nima/fullsystem_proto_logger*) » ls -l /opt                                                                                                                                                                                   nima@nima-MS-7C87
total 36
drwxr-xr-x  4 root root 4096 Aug 16  2022 cisco
drwxr-xr-x  8 root root 4096 Apr 30  2021 clion-2019.3.5
drwxr-xr-x  8 root root 4096 Jan  8  2022 clion-2021.2.3
drwxr-xr-x  7 root root 4096 Nov 26  2023 docker-desktop
drwxr-xr-x  3 root root 4096 Apr 30  2021 google
drwxr-xr-x  2 root root 4096 Jul 16  2021 jetbrains-toolbox-1.21.9547
drwxr-xr-x  3 root root 4096 Dec  9  2023 microsoft
drwxrwxrwx 10 root root 4096 Apr 30  2021 STM32CubeMX_6.2.0
drwxr-xr-x  8 nima nima 4096 May 22 20:42 tbotspython
```

I just tried it and I can save layouts (`ctrl + S` in thunderscope), but relaunching Thunderscope with it doesn't show anything from fullsystem anymore, though the layout is automatically loaded.

I guess if needed, we could update `setup_software.sh` as well.

Reply:
I looked into it and we already do `chown -R` in `setup_software.sh`. For what ever reason that did not work for you, or maybe when you were testing with upgrading to Python 3.10 you manually recreated the `/opt/tbotspython` python virtual env directory.


https://github.com/UBC-Thunderbots/Software/blob/00372dfe0534ab4465a506cdfd8382133c9469d5/environment_setup/setup_software.sh#L138-L138

Reply:
I think so. That could really be the case. Anyhow, I think I would just manually `chown` my tbotspython directory for now.

Reply:
Okay. Could you then update `DEFAULT_SAVE_DIRECTORY` to use `/opt` then?

Reply:
yeah, I will push an update now.